optimize/backend/src/test/java/io/camunda/optimize/service/util/BpmnModelUtilTest.java

```diff
@@ -0,0 +1,172 @@
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.camunda.bpm.model.bpmn.instance.EndEvent;
+import org.camunda.bpm.model.bpmn.instance.IntermediateCatchEvent;
+import org.camunda.bpm.model.bpmn.instance.Signal;
+import org.camunda.bpm.model.bpmn.instance.StartEvent;
+import org.camunda.bpm.model.xml.ModelInstance;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.junit.jupiter.api.Test;
+
+public class BpmnModelUtilTest {
+
+  public static final String CUSTOMER_ONBOARDING = "CustomerOnboarding";
+  public static final String END_EVENT_TYPE = "endEvent";
+  public static final String SERVICE_TASK_TYPE = "serviceTask";
+
+  public static final String SIMPLE_SERVICE_TASK_PROCESS =
+      Bpmn.convertToString(createSimpleServiceTaskProcess(CUSTOMER_ONBOARDING));
+
+  public static final String PROCESS_WITH_83_SIGNAL_EVENTS =
+      Bpmn.convertToString(createProcessWith83SignalEvents(CUSTOMER_ONBOARDING));
+
+  @Test
+  void shouldParseBpmnModel() {
+    // when
+    final BpmnModelInstance modelInstance = parseBpmnModel(SIMPLE_SERVICE_TASK_PROCESS);
+
+    // then
+    assertThat(modelInstance).isNotNull();
+  }
+
+  @Test
+  void shouldExtractFlowNodeData() {
+    // when
+    final List<FlowNodeDataDto> flowNodeData =
+        BpmnModelUtil.extractFlowNodeData(SIMPLE_SERVICE_TASK_PROCESS);
+
+    // then
+    assertThat(flowNodeData)
+        .isNotNull()
+        .isNotEmpty()
+        .extracting(FlowNodeDataDto::getId, FlowNodeDataDto::getType, FlowNodeDataDto::getName)
+        .containsExactlyInAnyOrder(
+            Tuple.tuple(START_EVENT, START_EVENT, START_EVENT),
+            Tuple.tuple(SERVICE_TASK, SERVICE_TASK_TYPE, SERVICE_TASK),
+            Tuple.tuple(END_EVENT, END_EVENT_TYPE, null));
+  }
+
+  @Test
+  void shouldExtractUserTaskNames() {
+    // when
+    final String bpmnModelInstance =
+        Bpmn.convertToString(createSimpleUserTaskProcess(CUSTOMER_ONBOARDING));
+    final Map<String, String> userTaskNames = BpmnModelUtil.extractUserTaskNames(bpmnModelInstance);
+
+    // then
+    assertThat(userTaskNames).isNotNull().isNotEmpty().containsKey(USER_TASK);
+  }
+
+  @Test
+  void shouldExtractProcessDefinitionName() {
+    // when
+    final Optional<String> processName =
+        BpmnModelUtil.extractProcessDefinitionName(
+            CUSTOMER_ONBOARDING, SIMPLE_SERVICE_TASK_PROCESS);
+
+    // then
+    assertThat(processName).isPresent().get().isEqualTo(CUSTOMER_ONBOARDING);
+  }
+
+  @Test
+  void shouldExtractFlowNodeNames() {
+    // when
+    final List<FlowNodeDataDto> flowNodeData =
+        BpmnModelUtil.extractFlowNodeData(SIMPLE_SERVICE_TASK_PROCESS);
+    final Map<String, String> flowNodeNames = BpmnModelUtil.extractFlowNodeNames(flowNodeData);
+
+    // then
+    assertThat(flowNodeNames)
+        .isNotNull()
+        .isNotEmpty()
+        .containsKeys(START_EVENT, SERVICE_TASK, END_EVENT)
+        .containsValues(START_EVENT, SERVICE_TASK, null);
```

Comment:
```suggestion
    assertThat(flowNodeNames)
        .isNotNull()
        .isNotEmpty()
        .containsExactlyEntriesOf(
          Map.ofEntries(
            entry(START_EVENT, START_EVENT),
            entry(SERVICE_TASK, SERVICE_TASK),
            entry(END_EVENT, null)
          )
        )
```

This is a stronger assertion as it guarantees that keys and values are bound correctly together

Reply:
fixed